### PR TITLE
[VM] cleanup some changes made in the TangerineWhistle PR

### PR DESCRIPTION
**Pull request by jochem-brouwer**
After reviewing #807 some code quality comments were made. This PR intends to resolve these comments.

### CODE REVIEW

**Comment by jochem-brouwer** [comment]
Why is my second commit not showing up? It is in the branch? https://github.com/ethereumjs/ethereumjs-vm/tree/cleanup-tangerine-whistle

**Comment by jochem-brouwer** [comment]
Found an issue with the TW implementation in BlockchainTests. Will push a fix soon which I'd like to have in this PR.

**Comment by jochem-brouwer** [comment]
Ready to review.

File: packages/vm/lib/state/cache.ts [hunk 1]
```diff
@@ -51,30 +51,39 @@ export default class Cache {
...
+    return account
   }
 
   /**
    * Looks up address in cache, if not found, looks it up
    * in the underlying trie.
    * @param key - Address of account
-   * @param create - Create emtpy account if non-existent
    */
-  async getOrLoad(key: Buffer, create: boolean = true): Promise<Account | undefined> {
```
Comment (ryanio) [inline packages/vm/lib/state/cache.ts:73]: looks good, does this mean the casting i was referencing in this comment can be removed then? https://github.com/ethereumjs/ethereumjs-vm/pull/807#discussion_r454045007

Comment (jochem-brouwer) [reply packages/vm/lib/state/cache.ts:73]: Forgot to remove this cast, thanks 😄

File: packages/vm/lib/evm/eei.ts [hunk 2]
```diff
@@ -583,6 +583,14 @@ export default class EEI {
     return this._state.accountIsEmpty(address)
   }
 
+  /**
+   * Returns true if account exists in the state trie (it can be empty). Returns false if the account is `null`.
+   * @param address - Address of account
+   */
+  async accountExists(address: Buffer): Promise<boolean> {
+    return this._state.accountExists(address)
```
Comment (evertonfraga) [inline packages/vm/lib/evm/eei.ts:591]: This is currently not covered by tests:
https://codecov.io/gh/ethereumjs/ethereumjs-vm/compare/2009f18c81f2fe253e846333ace57eb77625dd60...681418a97f5e27b56da0f5ca63a6fd4acb244c2b/src/packages/vm/lib/evm/eei.ts?before=packages/vm/lib/evm/eei.ts#L591

File: packages/vm/lib/state/cache.ts [hunk 3]
```diff
@@ -51,30 +51,39 @@ export default class Cache {
...
+   * Returns true if the key was deleted and thus existed in the cache earlier
+   * @param key - trie key to lookup
+   */
+  keyIsDeleted(key: Buffer): boolean {
+    const keyStr = key.toString('hex')
+    const it = this._cache.find(keyStr)
+    if (it.node) {
+      return it.value.deleted
+    }
+    return false
```
Comment (evertonfraga) [inline packages/vm/lib/state/cache.ts:64]: This is currently not covered by tests:
https://codecov.io/gh/ethereumjs/ethereumjs-vm/compare/2009f18c81f2fe253e846333ace57eb77625dd60...681418a97f5e27b56da0f5ca63a6fd4acb244c2b/src/packages/vm/lib/state/cache.ts?before=packages/vm/lib/state/cache.ts#L64

File: packages/vm/lib/evm/eei.ts (hunk 2 above)
Comment (jochem-brouwer) [reply packages/vm/lib/evm/eei.ts:591]: Added a test for this (`StateManager` accountExists is actually being tested; the `accountExists` of EEI is actually a proxy function but we should still test that)

File: packages/vm/lib/state/cache.ts (hunk 3 above)
Comment (jochem-brouwer) [reply packages/vm/lib/state/cache.ts:64]: Added test for this

**Comment by jochem-brouwer** [comment]
@evertonfraga The tests in 49f21ff should increase test coverage to your requested missing coverage 😄

**Pull request merged**